include/librealsense2/dds/dds-device-server.cpp

```diff
@@ -217,14 +217,14 @@ class dds_device_server::dds_notifications_server
         }
     };
 
-    void send_notification( const topics::raw::device::notification& msg ) 
+    void send_notification( const topics::raw::device::notification& notification ) 
     {
         std::unique_lock< std::mutex > lock( _notification_send_mutex );
-        topics::raw::device::notification msg_to_move( msg );
-        if( ! _instant_notifications.enqueue( std::move( msg_to_move ) ) )
+        if( ! _instant_notifications.enqueue(
+                std::move( const_cast< topics::raw::device::notification & >( notification ) ) ) )
```

Comment:
O_O
Danger Will Robinson

Reply:
What is the concern?
The `&` ?

I see the data dequeued as expected..
I wanted to save the extra assignment
Please elaborate..

Reply:
And a function that moves it's input parameter should signal it by receiving a `&&`. The caller should know not to use the variable again.

Reply:
Guy's you are right, I totally agree :)
`libRS` has a known missing API in `concurrency` helper classes were it doesn't have a `const` input API.
This caused me to drop the qualifier.
I took Ohad's advise and change the function input to get an R-value variable so the user will understand that this input should not be used after the calling.

Later on we need to see how we can expent the concurrency classes API to handle this cases.
Currently I want to wrap this PR so that the DDS work can continue on Ohad's side.

Fair enough?